/*
 * Copyright (c) 2018 Cisco Systems, Inc. and others.  All rights reserved.
 *
 * This program and the accompanying materials are made available under the
 * terms of the Eclipse Public License v1.0 which accompanies this distribution,
 * and is available at http://www.eclipse.org/legal/epl-v10.html
 *
 */

syntax = "proto3";

package openbmp;

// The BGP2BMP Service Interface.
service OPENBMPService {
    //The disable peering session RPC
    rpc DisablePeer (Ip) returns (Response) {}

    //The enable peering session RPC
    rpc EnablePeer (Ip) returns (Response) {}

    //The disable BMP router session RPC
    rpc DisableRouter (Ip) returns (Response) {}

    //The enable BMP router session RPC
    rpc EnableRouter (Ip) returns (Response) {}

}

// The request message containing the peer's Ip in printable format.
message Ip {
    string ip = 1;
}

// The response message
message Response {
    string message = 1;
}